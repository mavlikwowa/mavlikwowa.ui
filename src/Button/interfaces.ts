/* Libraries */
import React from 'react';

/* Sizes of button */
export enum ButtonSize {
  SMALL = 'SMALL',
  MEDIUM = 'MEDIUM',
  LARGE = 'LARGE',
}

/* Props of Button component */
export interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
  /* Settings of width (in spite of the size settings, changes the button`s width) */
  width?: string;
  /* A flag of loading (shows a preloader) */
  isLoading?: boolean;
  /* Settings of Background */
  background?: string;
  /* Size */
  buttonSize?: ButtonSize;
  /* Color */
  borderColor?: string;
  /* A showing element */
  // children: React.ReactNode;
}
