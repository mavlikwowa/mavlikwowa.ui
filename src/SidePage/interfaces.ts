/* Libraries */
import React from 'react';

export interface SidePageProps
  extends React.HTMLProps<HTMLDivElement>,
    SidePageContainerProps {
  /* a header`s element */
  header?: React.ReactNode;
  /* An action after click on Close Icon */
  onCloseClick: () => void;
}

export interface SidePageContainerProps {
  /* a showing flag */
  show: boolean;
  /* a color of background */
  background?: string;
}
