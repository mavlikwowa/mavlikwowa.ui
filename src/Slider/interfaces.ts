export interface SliderProps {
  data: React.ReactNode[];
  width?: string;
  height?: string;
}

export interface StyledSliderContentProps {
  slideAmount: number;
  // Important! We should calculate a content width
  width: number;
  height?: string;
}

export interface StyledSliderContainerProps {
  width?: string;
  height?: string;
}
