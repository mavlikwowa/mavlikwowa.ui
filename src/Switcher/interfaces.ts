export interface SwitcherProps extends React.HTMLProps<HTMLSpanElement> {
  hasChildren?: React.ReactNode;
  checked?: boolean;
}

export interface StyledSwitcherProps {
  hasChildren: boolean;
}
