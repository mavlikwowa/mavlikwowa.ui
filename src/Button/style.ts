/* Libraries */
import styled from '@emotion/styled';

/* Interfaces */
import { ButtonProps, ButtonSize } from './interfaces';

/* Styled component */
export const StyledButton = styled.button`
  ${({ width, background, buttonSize }: ButtonProps) => {
    // checks parametrs and creates sizes of button
    const createButtonSize = () => {
      const size = {
        width: 'auto',
        height: '50px',
      };
      if (width) {
        size.width = width;
      } else {
        switch (buttonSize) {
          case ButtonSize.SMALL:
            size.width = '150px';
            size.height = '50px';
            break;
          case ButtonSize.MEDIUM:
            size.width = '200px';
            size.height = '75px';
            break;
          case ButtonSize.LARGE:
            size.width = '250px';
            size.height = '100px';
            break;
          default:
            break;
        }
      }
      return size;
    };
    return `
      display: flex;
      align-items: baseline;
      justify-content: center;
      width: ${createButtonSize().width};
      height: ${createButtonSize().height};
      max-width: 250px;
      background: ${background || 'auto'};
    `;
  }}
`;
