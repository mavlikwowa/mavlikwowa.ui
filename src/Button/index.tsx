/* Libraries */
import React from 'react';
/* Styles */
import { StyledButton } from './style';
/* Interfaces */
import { ButtonProps } from './interfaces';

/* Button component */
const Button: React.FC<ButtonProps> = (
  {
    isLoading,
    children,
    // should redefine this types, because React.HTMLProps<HTMLButtonElement> has incompatible similar types
    type = 'button',
    as = undefined,
    ...props
  }: ButtonProps,
  ...others
) => {
  return (
    <StyledButton {...props} {...others}>
      {isLoading ? 'Loading...' : children}
    </StyledButton>
  );
};

export default Button;
