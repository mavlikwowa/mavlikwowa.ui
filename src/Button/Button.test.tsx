/* Libraries */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

/* Components */
import Button from './index';

/* Interfaces */
import { ButtonProps, ButtonSize } from './interfaces';

/** Setups */
const setUp = () => {
  /* A test`s data */
  const props: ButtonProps = {
    buttonSize: ButtonSize.SMALL,
    children: 'Button text',
  };
  /* Component */
  const wrapper = shallow(<Button {...props}>{props.children}</Button>);
  return { props, wrapper };
};

/* Tests */
describe('Test component Button', () => {
  let wrapper: ShallowWrapper;
  let props: ButtonProps;

  beforeEach(() => {
    ({ wrapper, props } = setUp());
  });

  afterEach(() => {
    if (!wrapper) return;
    wrapper.unmount();
  });

  it('should render a component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should find a text of button', () => {
    expect(wrapper.text()).toBe(props.children);
  });
});
