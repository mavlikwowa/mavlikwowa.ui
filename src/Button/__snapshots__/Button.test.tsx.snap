// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Button should render a component 1`] = `
<Styled(button)
  0={Object {}}
  buttonSize="SMALL"
>
  Button text
</Styled(button)>
`;
